codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,253 @@
+
+    private static final String[] DEFAULT_PRELOAD_ERRORS = {
+            "ERR",
+            "ERR index out of range",
+            "ERR no such key",
+            "ERR source and destination objects are the same",
+            "ERR syntax error",
+            "BUSY Redis is busy running a script. You can only call SCRIPT KILL or SHUTDOWN NOSAVE.",
+            "BUSYKEY Target key name already exists.",
+            "EXECABORT Transaction discarded because of previous errors.",
+            "LOADING Redis is loading the dataset in memory",
+            "MASTERDOWN Link with MASTER is down and slave-serve-stale-data is set to 'no'.",
+            "MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk. " +
+                    "Commands that may modify the data set are disabled. Please check Redis logs for details " +
+                    "about the error.",
+            "NOAUTH Authentication required.",
+            "NOREPLICAS Not enough good slaves to write.",
+            "NOSCRIPT No matching script. Please use EVAL.",
+            "OOM command not allowed when used memory > 'maxmemory'.",
+            "READONLY You can't write against a read only slave.",
+            "WRONGTYPE Operation against a key holding the wrong kind of value",
+    };
+
+    private static PreloadStringsCache preloadSimpleString;
+    private static PreloadStringsCache preloadErrors;
+
+    static {
+        setPreloadStrings(RedisMessageType.SIMPLE_STRING, DEFAULT_PRELOAD_SIMPLE_STRINGS);
+        setPreloadStrings(RedisMessageType.ERROR, DEFAULT_PRELOAD_ERRORS);
+    }
+
+    private RedisMessageFactory() {
+    }
+
+    /**
+     * Set preload strings. Users can override preload strings before using {@link RedisMessageFactory}.
+     *
+     * @param type one of {@link RedisMessageType#SIMPLE_STRING} or {@link RedisMessageType#ERROR}.
+     * @param messages an array of String to preload.
+     */
+    public static void setPreloadStrings(RedisMessageType type, String... messages) {
+        ObjectUtil.checkNotNull(messages, "messages");
+        if (type == RedisMessageType.SIMPLE_STRING) {
+            preloadSimpleString = convertPreloadsCache(RedisMessageType.SIMPLE_STRING, messages);
+        } else if (type == RedisMessageType.ERROR) {
+            preloadErrors = convertPreloadsCache(RedisMessageType.ERROR, messages);
+        } else {
+            throw new IllegalArgumentException("not support type: " + type);
+        }
+    }
+
+    private static PreloadStringsCache convertPreloadsCache(RedisMessageType type, String... messages) {
+        Map<String, RedisMessage> stringsCache = new HashMap<String, RedisMessage>();
+        Map<ByteBuf, RedisMessage> byteBufCache = new HashMap<ByteBuf, RedisMessage>();
+        for (String message : messages) {
+            ByteBuf byteBuf = Unpooled.wrappedBuffer(message.getBytes(CharsetUtil.UTF_8));
```

Comment:
Users cannot read/consume this content directly because it's not open to public and used by only key of `byteBufCache`. Anyway, I'll change it to `unmodifiableBuffer`.